persistence/test/ibc_test.go

```diff
@@ -120,3 +121,180 @@ func TestGetIBCStoreEntry(t *testing.T) {
 		})
 	}
 }
+
+type attribute struct {
+	key   []byte
+	value []byte
+}
+
+func TestIBCSetEvent(t *testing.T) {
+	db := NewTestPostgresContext(t, 1)
+	event := new(coreTypes.IBCEvent)
+	event.Topic = "test"
+	event.Height = 1
+	event.Attributes = append(event.Attributes, &coreTypes.Attribute{
+		Key:   []byte("testKey"),
+		Value: []byte("testValue"),
+	})
+	require.NoError(t, db.SetIBCEvent(event))
+
+	testCases := []struct {
+		name        string
+		height      uint64
+		topic       string
+		attributes  []attribute
+		expectedErr string
+	}{
+		{
+			name:   "Successfully set new event at height 1",
+			height: 1,
+			topic:  "test",
+			attributes: []attribute{
+				{
+					key:   []byte("key"),
+					value: []byte("value"),
+				},
+				{
+					key:   []byte("key2"),
+					value: []byte("value2"),
+				},
+			},
+			expectedErr: "",
+		},
+		{
+			name:   "Successfully set new event at height 2",
+			height: 2,
+			topic:  "test",
+			attributes: []attribute{
+				{
+					key:   []byte("key"),
+					value: []byte("value"),
+				},
+				{
+					key:   []byte("key2"),
+					value: []byte("value2"),
+				},
+			},
+			expectedErr: "",
+		},
+		{
+			name:   "Successfully set a duplicate event new height",
+			height: 2,
+			topic:  "test",
+			attributes: []attribute{
+				{
+					key:   []byte("testKey"),
+					value: []byte("testValue"),
+				},
+			},
+			expectedErr: "",
+		},
+		{
+			name:   "Fails to set a duplicate event at height 1",
+			height: 1,
+			topic:  "test",
+			attributes: []attribute{
+				{
+					key:   []byte("testKey"),
+					value: []byte("testValue"),
+				},
+			},
+			expectedErr: "ERROR: duplicate key value violates unique constraint \"ibc_events_pkey\" (SQLSTATE 23505)",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			db.Height = int64(tc.height)
+			event := new(coreTypes.IBCEvent)
+			event.Topic = tc.topic
+			event.Height = tc.height
+			for _, attr := range tc.attributes {
+				event.Attributes = append(event.Attributes, &coreTypes.Attribute{
+					Key:   attr.key,
+					Value: attr.value,
+				})
+			}
+			err := db.SetIBCEvent(event)
+			if tc.expectedErr != "" {
+				require.EqualError(t, err, tc.expectedErr)
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestGetIBCEvent(t *testing.T) {
+	db := NewTestPostgresContext(t, 1)
+	events := make([]*coreTypes.IBCEvent, 0, 4)
```

Comment:
Leave a comment explaining the preparation you're doing to the state for the tests below